### PR TITLE
Add code generator for Settings.

### PR DESCRIPTION
**Pull request by prideout**
This adds a C++ code generator, implemented in Go. For now this only
generates the JSON reader / writer code.

This PR also prepares `Options.h` (the ground truth) by simplifying its
syntax just a bit. The ground truth file must have very simple C++
syntax, which is described in the README:

https://github.com/google/filament/blob/pr/codegen2/tools/codegen-options/README.md

This process revealed a small bug: `Filter.MEDIAN` was bound to the
incorrect value in Java.

Still TBD is the codegen for Java and JavaScript bindings.

### CODE REVIEW

File: tools/codegen-options/main.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
+/*
```
Comment (bejado) [inline tools/codegen-options/main.go:1]: Tabs 😱  is that a Go requirement?

**Review by bejado — COMMENTED** [review]
Our first Go contribution!

File: tools/codegen-options/main.go (hunk 1 above)
Comment (prideout) [reply tools/codegen-options/main.go:1]: LOL yup, the language comes with an opinionated format via the gofmt tool, which uses tabs for indention and spaces for various alignment things, e.g. making struct fields line up.

**Review by bejado — APPROVED** [review]
Unfamiliar with Go, but LGTM!

**Pull request merged**